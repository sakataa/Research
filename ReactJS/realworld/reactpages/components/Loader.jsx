import React from 'react';
import ReactDOM from 'react-dom';

export default class Loader extends React.Component {
    constructor(props) {
        super(props);
    };
    render() {
        return <div className="loading-process default" style={{ display: `${this.props.isLoading ? 'block' : 'none'}` }}>
            <div className="icon-loading">
                <div className="loader-inner ball-beat">
                    <div></div>
                    <div></div>
                    <div></div>
                </div>
            </div>
        </div>
    }
};
