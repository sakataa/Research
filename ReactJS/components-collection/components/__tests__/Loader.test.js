import React from 'react'
import Loader from '../Loader'
import renderer from 'react-test-renderer'

describe('Loader ', () => {
    it('isLoading = false - component render correctly', () => { 
        const props = {
            isLoading: false
        };

        const component = renderer.create(<Loader {...props} />);

        const result = component.toJSON(); 
        
        expect(result).toMatchSnapshot(); 
    });

    it('isLoading = true - component render correctly', () => {
        const props = {
            isLoading: true
        };

        const component = renderer.create(<Loader {...props} />); 

        const result = component.toJSON(); 

        expect(result).toMatchSnapshot(); 
    });
})