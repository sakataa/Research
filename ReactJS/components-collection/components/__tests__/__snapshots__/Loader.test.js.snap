// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader  isLoading = false - component render correctly 1`] = `
<div
  className="loading-process default"
  style={
    Object {
      "display": "none",
    }
  }
>
  <div
    className="icon-loading"
  >
    <div
      className="loader-inner ball-beat"
    >
      <div />
      <div />
      <div />
    </div>
  </div>
</div>
`;

exports[`Loader  isLoading = true - component render correctly 1`] = `
<div
  className="loading-process default"
  style={
    Object {
      "display": "block",
    }
  }
>
  <div
    className="icon-loading"
  >
    <div
      className="loader-inner ball-beat"
    >
      <div />
      <div />
      <div />
    </div>
  </div>
</div>
`;
